Add Crossplane 2023 security audit report

AdamKorcz opened the pull request:
<!--
Thank you for helping to improve Crossplane!

Please read through https://git.io/fj2m9 if this is your first time opening a
Crossplane pull request. Find us in https://slack.crossplane.io/messages/dev if
you need any help contributing.
-->

### Description of your changes

<!--
Briefly describe what this pull request does, and how it is covered by tests.
Be proactive - direct your reviewers' attention to anything that needs special
consideration.

You MUST either [x] check or ~strikethrough~ every item in the checklist below.

We love pull requests that fix an open issue. If yours does, use the below line
to indicate which issue it fixes, for example "Fixes #500".
-->

Adds the 2023 Crossplane security audit report. 

Thank you to everyone involved!

@phisco @jbw976 @negz @turkenh @jeanduplessis-alt @davidkorczynski 

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Added or updated unit **and** E2E tests for my change.
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR if necessary.

[contribution process]: https://git.io/fj2m9

phisco reviewed (approved):
Minor: not sure why but there is an highlighted "Fixed in: ..." at some point in the pdf.
<img width="982" alt="image" src="https://github.com/crossplane/crossplane/assets/5697904/6ee0e754-4f3b-42ce-b3db-5de6895c81f0">

Otherwise, LGTM!

Thanks @AdamKorcz, @DavidKorczynski and the rest of the team for your work!

AdamKorcz commented:
> Minor: not sure why but there is an highlighted "Fixed in: ..." at some point in the pdf. <img alt="image" width="982" src="https://user-images.githubusercontent.com/5697904/256539409-6ee0e754-4f3b-42ce-b3db-5de6895c81f0.png">
> 
> Otherwise, LGTM!
> 
> Thanks @AdamKorcz, @DavidKorczynski and the rest of the team for your work!

Thank you, @phisco. Fixed and rebased.
